src/core/tests/Test_YumPackageManager.py

```diff
@@ -648,6 +655,244 @@ def test_update_image_default_patch_mode_raises_exception(self):
         self.runtime.env_layer.file_system.write_with_retry = self.mock_write_with_retry_raise_exception
         self.assertRaises(Exception, package_manager.update_os_patch_configuration_sub_setting)
 
+    def test_revert_auto_os_update_to_system_default_success_with_all_services_enabled(self):
+        self.runtime.set_legacy_test_type('HappyPath')
+        package_manager = self.container.get('package_manager')
+
+        # setup current auto OS update config
+        package_manager.yum_cron_configuration_settings_file_path = os.path.join(self.runtime.execution_config.config_folder, "yum-cron.conf")
+        yum_cron_os_patch_configuration_settings = 'apply_updates = no\ndownload_updates = no\n'
+        self.runtime.write_to_file(package_manager.yum_cron_configuration_settings_file_path, yum_cron_os_patch_configuration_settings)
+
+        package_manager.dnf_automatic_configuration_file_path = os.path.join(self.runtime.execution_config.config_folder, "automatic.conf")
+        dnf_automatic_os_patch_configuration_settings = 'apply_updates = no\ndownload_updates = no\n'
+        self.runtime.write_to_file(package_manager.dnf_automatic_configuration_file_path, dnf_automatic_os_patch_configuration_settings)
+
+        package_manager.packagekit_configuration_file_path = os.path.join(self.runtime.execution_config.config_folder, "PackageKit.conf")
+        packagekit_os_patch_configuration_settings = 'WritePreparedUpdates = false\nGetPreparedUpdates = false\n'
+        self.runtime.write_to_file(package_manager.packagekit_configuration_file_path, packagekit_os_patch_configuration_settings)
+
+        # setup backup for system default auto OS update config
+        package_manager.image_default_patch_configuration_backup_path = os.path.join(self.runtime.execution_config.config_folder, Constants.IMAGE_DEFAULT_PATCH_CONFIGURATION_BACKUP_PATH)
+        backup_image_default_patch_configuration_json = {
+            "yum-cron": {
+                "apply_updates": "yes",
+                "download_updates": "yes",
+                "enable_on_reboot": True,
+                "installation_state": True
+            },
+            "dnf-automatic": {
+                "apply_updates": "yes",
+                "download_updates": "yes",
+                "enable_on_reboot": True,
+                "installation_state": True
+            },
+            "packagekit": {
+                "WritePreparedUpdates": "true",
+                "GetPreparedUpdates": "true",
+                "enable_on_reboot": True,
+                "installation_state": True
+            }
+        }
```

Comment:
Many examples of de-dupable code in tests here.

The tests also read easier when the common set up portions and common assert portions are parameterized into private functions.

Reply:
Refactored